diff --git a/dist/esm/server/base-server.js b/dist/esm/server/base-server.js
index b4e570e50ecc34ffa02f727a37812b1863b8e954..1efa1e1f08f9f68711a8c0c4c58a09eeb3b5481e 100644
--- a/dist/esm/server/base-server.js
+++ b/dist/esm/server/base-server.js
@@ -390,7 +390,7 @@ export default class Server {
             // ensure cookies set in middleware are merged and
             // not overridden by API routes/getServerSideProps
             const _res = res.originalResponse || res;
-            const origSetHeader = _res.setHeader.bind(_res);
+            const origSetHeader = _res && typeof _res.setHeader === 'function' ? _res.setHeader.bind(_res) : null
             _res.setHeader = (name, val)=>{
                 // When renders /_error after page is failed,
                 // it could attempt to set headers after headers
diff --git a/dist/server/base-server.js b/dist/server/base-server.js
index ff366057ee10a37eba453731597a415c46872493..7997a764a84920a481f72ab12887c264e7ce4e56 100644
--- a/dist/server/base-server.js
+++ b/dist/server/base-server.js
@@ -457,7 +457,7 @@ class Server {
             // ensure cookies set in middleware are merged and
             // not overridden by API routes/getServerSideProps
             const _res = res.originalResponse || res;
-            const origSetHeader = _res.setHeader.bind(_res);
+            const origSetHeader = _res && typeof _res.setHeader === 'function' ? _res.setHeader.bind(_res) : null
             _res.setHeader = (name, val)=>{
                 // When renders /_error after page is failed,
                 // it could attempt to set headers after headers
